src/librustc_trans/trans/context.rs

```diff
@@ -893,10 +891,11 @@ fn declare_intrinsic(ccx: &CrateContext, key: & &'static str) -> Option<ValueRef
         )
     }
 
-    compatible_ifn!("llvm.copysign.f32", copysignf(t_f32, t_f32) -> t_f32);
-    compatible_ifn!("llvm.copysign.f64", copysign(t_f64, t_f64) -> t_f64);
-    compatible_ifn!("llvm.round.f32", roundf(t_f32) -> t_f32);
-    compatible_ifn!("llvm.round.f64", round(t_f64) -> t_f64);
+    compatible_ifn!("llvm.copysign.f32", copysignf(t_f32, t_f32) -> t_f32, 4);
+    compatible_ifn!("llvm.copysign.f64", copysign(t_f64, t_f64) -> t_f64, 4);
+    compatible_ifn!("llvm.round.f32", roundf(t_f32) -> t_f32, 4);
+    compatible_ifn!("llvm.round.f64", round(t_f64) -> t_f64, 4);
+    compatible_ifn!("llvm.assume", fn(i1) -> void, 6);
```

Comment:
I suspect this isn't quite right, unfortunately; it seems like it might be declaring a substitute C function with signature `void fn(i1)`. (I.e. the `compatible_ifn!` macro has a different syntax.)